Add GreenPark Sports and Fauna case studies (#913)
pulumi/pulumi-hugo: themes/default/content/case-studies/greenpark-sports.md

---
title: GreenPark Sports
description: |

>     Pulumi enabled GreenPark Sports to empower its developers with self-service cloud infrastructure, enabling them to quickly and easily deploy Kubernetes clusters and contribute infrastructure code.

Reviewer: ```suggestion
    Pulumi enabled GreenPark Sports to empower its developers with self-service cloud infrastructure, allowing them to quickly and easily deploy Kubernetes clusters and contribute infrastructure code.
```

---
title: GreenPark Sports
description: |

>     Pulumi enabled GreenPark Sports to empower its developers with self-service cloud infrastructure through a “developer-first” approach.

Reviewer: ```suggestion
    Pulumi enabled GreenPark Sports to empower its developers with self-service cloud infrastructure, enabling them to quickly and easily deploy Kubernetes clusters and contribute infrastructure code.
```

Pulumi would also enable Jacob to build, share, and reuse infrastructure code with the full benefits of a software supply chain. His team would be able to build higher-level components that make it easier for developers to build production-quality cloud applications. The components would be wrapper functions and helper functions written in Go, which developers could use to build and deploy a fully-configured Kubernetes application.

When recounting how he first started evaluating Pulumi, Jacob said, “I took a look at Pulumi and that started a much happier time in my life.”

### Putting "Developers First" with Cloud Infrastructure

After deciding on Pulumi, Jacob and his team built a system based on a “developer-first” approach. The system made developers as independent as possible by allowing them to provision cloud infrastructure, with guardrails to maintain security.

#### Migrating Infrastructure to Pulumi

> First, the platform team migrated its infrastructure from Terraform to Pulumi. This included Kubernetes resources, Google Cloud resources, GitHub repos, DataDog, CloudFlare, Confluent for Kafka, Consul, and Vault. This enabled them to build, deploy, and manage all of their cloud and SaaS resources from a single platform, using Go as their language of choice.

Reviewer: ```suggestion
First, the platform team [migrated](/tf2pulumi/) its infrastructure from Terraform to Pulumi. This included Kubernetes resources, Google Cloud resources, GitHub repos, DataDog, CloudFlare, Confluent for Kafka, Consul, and Vault. This enabled them to build, deploy, and manage all of their cloud and SaaS resources from a single platform, using Go as their language of choice.
```

<img class="block mx-auto md:max-w-4xl my-8" src="/images/case-studies/greenpark-sports-monorepo-diagram.png">
<p class="text-sm italic text-center">GreenPark Sports uses a monorepo to manage its infrastructure code.</p>

#### Simplifying Kubernetes Deployments

> Pulumi also simplified how they manage Kubernetes deployments. It gave them a strongly-typed way of modeling Kubernetes resources with standard languages (e.g., Go) instead of YAML. Using Pulumi’s native support for Kubernetes, they could access 100% of Kubernetes’ APIs. They could also build and manage Helm charts and Kubernetes custom resource definitions with Pulumi’s cloud native integrations.

Reviewer: ```suggestion
Pulumi also simplified how they manage Kubernetes deployments. It gave them a strongly-typed way of modeling Kubernetes resources with standard languages (e.g., Go) instead of YAML. Using Pulumi’s [native support for Kubernetes](/kubernetes), they could access 100% of Kubernetes’ APIs. They could also build and manage Helm charts and Kubernetes custom resource definitions with Pulumi’s cloud native integrations.
```

#### Building Reusable Cloud Infrastructure

> Next, Jacob’s team built a set of reusable Pulumi components in Go which developers could use to build and deploy new services without having to think about infrastructure configurations. Components abstract away infrastructure complexity, making them easy to consume. For example, Jacob’s team built a component in 8 lines of code representing a service with around 100 resources configured with best practices. When using the component, developers only needed to input a name for the service, the environment they’re running in, and a few other settings.

Reviewer: ```suggestion
Next, Jacob’s team built a set of [reusable Pulumi components](/docs/intro/concepts/resources/#components) in Go which developers could use to build and deploy new services without having to think about infrastructure configurations. Components abstract away infrastructure complexity, making them easy to consume. For example, Jacob’s team built a component in 8 lines of code representing a service with around 100 resources configured with best practices. When using the component, developers only needed to input a name for the service, the environment they’re running in, and a few other settings.
```

#### Empowering Developers to Self-Service with Infrastructure

> Finally, Jacob’s team established a workflow that allows developers to independently deploy infrastructure. It also ensures that changes comply with company standards and security. In this workflow, all changes are managed through GitHub pull requests (PRs). If someone wants to deploy something new or an update, they create a PR and the platform team reviews the request. Using the Pulumi Console, platform team members can visualize what resources will be created or changed. They can also view logs of past changes. If the PR is approved and merged, then the changes are built, tested, and deployed with GreenPark’s CI/CD workflow, which uses Flux CD and Bazel.

Reviewer: ```suggestion
Finally, Jacob’s team established a workflow that allows developers to independently deploy infrastructure. It also ensures that changes comply with company standards and security. In this workflow, all changes are managed through GitHub pull requests (PRs). If someone wants to deploy something new or an update, they create a PR and the platform team reviews the request. Using the Pulumi Service console, platform team members can visualize what resources will be created or changed. They can also view logs of past changes. If the PR is approved and merged, then the changes are built, tested, and deployed with GreenPark’s CI/CD workflow, which uses Flux CD and Bazel.
```

### How Pulumi Benefits GreenPark Sports

Pulumi enabled GreenPark Sports to empower its developers with self-service cloud infrastructure through a “developer-first” approach. The benefits were:

- Grew Infrastructure as Code adoption from <1/3 of engineers to 100%, which is defined as anyone who makes changes to GreenPark’s infrastructure codebase.
- Increased the number of infrastructure deployments initiated by developers by 70% since adopting Pulumi.
- The platform team empowered developers to build and deploy cloud infrastructure on a self-service basis while still enforcing security and compliance.
- Everyone can build, deploy, and manage cloud infrastructure, Kubernetes, and SaaS providers with general-purpose languages (Go, TypeScript/JavaScript, Python, C#) and existing development tools (e.g. IDEs, test frameworks, package managers).
- Simplified Kubernetes deployments because the platform team can manage Kubernetes resources, custom resources, and Helm charts from a single platform using general-purpose languages like Go.

> Want to work at GreenPark Sports? <a href="https://boards.greenhouse.io/greenparksports" _target="_blank" rel="noopener noreferrer">They're hiring!</a>

Reviewer: remove this line

Author: 👍
